overlord/snapstate/check_snap.go

```diff
@@ -482,13 +482,11 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, _ snap.C
 	if err == state.ErrNoState {
 		// check if we are in the remodel case
 		if deviceCtx != nil && deviceCtx.ForRemodeling() {
-			model := deviceCtx.Model()
-			if model.Kernel() == snapInfo.InstanceName() {
+			if whichName(deviceCtx.Model()) == snapInfo.InstanceName() {
 				return nil
 			}
 		}
-
-		return fmt.Errorf("internal error: cannot remodel gadget yet")
+		return fmt.Errorf("internal error: unknown device snap %q", snapInfo.InstanceName())
```

Comment:
I couldn't find "device snap" term in the existing code, is this an established term we're going to use?

Also, should we have a test for it in check_snap_test.go?

Reply:
I've tweaked the message and added a test that triggers the error.